src/util/dstring.h

```diff
@@ -158,6 +158,12 @@ inline size_t hash_string(const dstringt &s)
   return s.hash();
 }
 
+inline size_t const_hash_string(const dstringt &s)
+{
+  return hash_string(string_container.get_string(s.get_no()));
+}
```

Comment:
@martin-cs, @tautschnig: There is a need for us to identify goto-program hashes across commits (of the target software) in the case of differential analysis (specifically, we record hash-tagged information from an analysis and need to relate it to a new analysis done after some changes to the code). The current hash uses the id no. which can change from build to build while we actually hash the string.